src/js/common/schemaform/definitions/year.js

```diff
@@ -0,0 +1,17 @@
+import { validateYear } from '../validation';
+import TextWidget from '../widgets/TextWidget.jsx';
+import * as ReviewWidget from '../review/widgets.jsx';
+
+const uiSchema = {
+  'ui:title': 'Year',
+  'ui:widget': TextWidget,
+  'ui:reviewWidget': ReviewWidget.TextWidget,
+  'ui:validations': [
+    validateYear
+  ],
+  'ui:errorMessages': {
+    pattern: 'Please provide a valid year'
```

Comment:
I don't think this will ever be used, since this is an integer field and we don't set `pattern` schema properties for those. You could change `pattern` to `type`, since we that could theoretically be triggered.